### PR TITLE
feat(UIS-25): Get stock list by quantity sold by client

### PR DESCRIPTION
**Pull request by StaceyJin**
JIRA: https://champlainsaintlambert.atlassian.net/browse/UIS-25

Implementation of retrieving stock items by quantity sold by client Id completed for BACKEND:

- method added to OrderService, OrderServiceImpl and OrderRepository;
- method added to controller;
- relevant integration test added.

### CODE REVIEW

File: src/main/java/com/um/inventoryservice/BusinessLayer/DataSetupService.java [hunk 1]
```diff
@@ -29,16 +29,47 @@ public void run(String... args) throws Exception {
...
+        OrderDTO o6 = new OrderDTO("5", new ArrayList<>());
+        OrderDTO o7 = new OrderDTO("6", new ArrayList<>());
+        OrderDTO o8 = new OrderDTO("7", new ArrayList<>());
+        OrderDTO o9 = new OrderDTO("5", new ArrayList<>());
+        OrderDTO o10 = new OrderDTO("5", new ArrayList<>());
         StockOrderDTO so1 = new StockOrderDTO("2454544", "this is the test for stock item 1", 6);
         StockOrderDTO so2 = new StockOrderDTO("7486504", "this is the test for stock item 2", 7);
         StockOrderDTO so3 = new StockOrderDTO("9735693", "this is the test for stock item 3", 8);
         StockOrderDTO so4 = new StockOrderDTO("7486504", "this is the test for stock item 3 x2", 3);
+        StockOrderDTO so5 = new StockOrderDTO("2454544", "this is the test for stock item 5", 15);
```
Comment (Kam1st) [inline src/main/java/com/um/inventoryservice/BusinessLayer/DataSetupService.java:43]: we can now put data that will represent real-life situations for the system.

File: src/test/java/com/um/inventoryservice/PresentationLayer/OrderControllerIntegrationTest.java [hunk 2]
```diff
@@ -133,6 +134,27 @@ void getStockOrdersByQuantity() {
...
+                .verifyComplete();
+
+        webTestClient.get().uri("/orders/297445493/quantity")
+                .exchange()
+                .expectStatus().isOk()
+                .expectBodyList(StockOrderDTO.class)
+                .hasSize(1)
+                .consumeWith(stockList -> {
+                    List<StockOrderDTO> orders = stockList.getResponseBody();
+                    assertThat(orders).extracting("quantity").contains(6);
```
Comment (Kam1st) [inline src/test/java/com/um/inventoryservice/PresentationLayer/OrderControllerIntegrationTest.java:154]: what is extracting exactly?

**Review by Kam1st — APPROVED** [review]

File: src/test/java/com/um/inventoryservice/PresentationLayer/OrderControllerIntegrationTest.java (hunk 2 above)
Comment (StaceyJin) [reply src/test/java/com/um/inventoryservice/PresentationLayer/OrderControllerIntegrationTest.java:154]: Retrieving the value of the indicated attribute.

File: src/main/java/com/um/inventoryservice/BusinessLayer/DataSetupService.java (hunk 1 above)
Comment (StaceyJin) [reply src/main/java/com/um/inventoryservice/BusinessLayer/DataSetupService.java:43]: Will update the sample data at next opportunity.

**Pull request merged**